tests/integration/index/query_with_relation_filter_test.go

```diff
@@ -910,3 +910,72 @@ func TestQueryWithIndexOnOneToMany_IfIndexedRelationIsNil_EqNilFilterShouldUseIn
 
 	testUtils.ExecuteTestCase(t, test)
 }
+
+// This test was added during https://github.com/sourcenetwork/defradb/issues/2862
+// multiple indexed fields on the second object are required for the failure.
+func TestQueryWithIndexOnManyToOne_MultipleViaOneToMany(t *testing.T) {
+	test := testUtils.TestCase{
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type User {
+						name: String
+						devices: [Device]
+					}
+
+					type Device {
+						model: String
+						owner: User @index
+						manufacturer: Manufacturer @index
+					}
+
+					type Manufacturer {
+						name: String
+						devices: [Device]
+					}
+				`,
+			},
+			testUtils.CreateDoc{
+				DocMap: map[string]any{
+					"name": "John",
+				},
+			},
+			testUtils.CreateDoc{
+				CollectionID: 2,
+				DocMap: map[string]any{
+					"name": "Apple",
+				},
+			},
+			testUtils.CreateDoc{
+				CollectionID: 1,
+				DocMap: map[string]any{
+					"model":        "MacBook Pro",
+					"owner":        testUtils.NewDocIndex(0, 0),
+					"manufacturer": testUtils.NewDocIndex(2, 0),
+				},
+			},
+			testUtils.Request{
+				Request: `query {
+					User {
+						devices {
+							owner_id
+							manufacturer_id
+						}
+					}
+				}`,
+				Results: []map[string]any{
+					{
+						"devices": []map[string]any{
+							{
+								"owner_id":        "bae-1ef746f8-821e-586f-99b2-4cb1fb9b782f",
```

Comment:
note: if you wait for https://github.com/sourcenetwork/defradb/pull/2871 to merge you will be able to use `testUtils.NewDocIndex` for these too.

Reply:
Yeah, I might do that, then you can merge before you log off - the CI runners seem to be in short supply atm though :(

Reply:
no worries, if you merge first, I can change your asserts in my PR